### PR TITLE
Decouple plugin manager from libcontainerd package

### PR DESCRIPTION
**Pull request by cpuguy83**
libcontainerd has a bunch of platform dependent code and huge interfaces
that are a pain implement.
To make the plugin manager a bit easier to work with, extract the plugin
executor into an interface and move the containerd implementation to a
separate package.

### CODE REVIEW

**Review by thaJeztah — COMMENTED** [review]
:heart: love narrower interfaces

**Comment by thaJeztah** [comment]
Golint is failing 😅

```
21:27:27 Errors from golint:
21:27:27 plugin/manager.go:67:6: exported type ExecutorCreator should have comment or be unexported
21:27:27 plugin/manager.go:143:1: comment on exported method Manager.HandleExitEvent should be of the form "HandleExitEvent ..."
21:27:27 
21:27:27 Please fix the above errors. You can test via "golint" and commit the result.

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Comment by cpuguy83** [comment]
The error is legit, looking into it.

**Comment by cpuguy83** [comment]
Ok, experimental failure just looks like a pull issue with hub. It's just stuck while pulling. Restarting.

**Comment by vieux** [comment]
@cpuguy83 it doesn't compile on windows

**Comment by thaJeztah** [comment]
ping @vieux it's green now, PTAL 👍

**Comment by liron-l** [comment]
@cpuguy83 is this pull request planned to be merged? 
It will enable to easily reference `PluginStore` in swarmkit without bringing lots of unnecessary dependencies. Thanks.

**Comment by thaJeztah** [comment]
ping @cpuguy83 this needs a rebase

**Comment by cpuguy83** [comment]
Rebased

**Comment by vieux** [comment]
LGTM

File: plugin/manager.go [hunk 1]
```diff
@@ -133,42 +145,37 @@ func (pm *Manager) tmpDir() string {
...
+// HandleExitEvent is called when the executor receives the exit event
+// In the future we may change this, but for now all we care about is the exit event.
+func (pm *Manager) HandleExitEvent(id string) error {
+	p, err := pm.config.Store.GetV2Plugin(id)
+	if err != nil {
+		return err
+	}
 
-		os.RemoveAll(filepath.Join(pm.config.ExecRoot, id))
+	os.RemoveAll(filepath.Join(pm.config.ExecRoot, id))
```
Comment (dnephin) [inline plugin/manager.go]: `os.RemoveAll()` returns an error, maybe `logrus.Warn()` on error? This looks like it was an existing issue, so not blocking

**Review by dnephin — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
all green - merging :+1:

**Pull request merged**